### PR TITLE
(7/7) RFC79: Implement incremental upload of CNA segmented data

### PR DESCRIPTION
**Pull request by forus**
Reminder: After merging this PR, incorporate all feedback in a new PR.
After it's merged, create `rfc79` -> `main` PR and remove "do not merge" from this one https://github.com/cBioPortal/cbioportal/pull/10816

_Please note, to avoid the costly process of rebasing all pull requests (PRs), all feedback will be incorporated into the final PR._

### CODE REVIEW

File: src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java [hunk 1]
```diff
@@ -135,13 +144,13 @@ public void run() {
...
-            if (segmentDataExistsForCancerStudy(cancerStudy)) {
+            if (!updateMode && segmentDataExistsForCancerStudy(cancerStudy)) {
                  throw new IllegalArgumentException("Seg data for cancer study " + cancerStudy.getCancerStudyStableId() + " has already been imported: " + dataFile);
             }
         
             importCopyNumberSegmentFileMetadata(cancerStudy, properties);
             importCopyNumberSegmentFileData(cancerStudy, dataFile);
-            DaoCopyNumberSegment.createFractionGenomeAlteredClinicalData(cancerStudy.getInternalId());
+            DaoCopyNumberSegment.createFractionGenomeAlteredClinicalData(cancerStudy.getInternalId(), processedSampleIds, updateMode);
             if( MySQLbulkLoader.isBulkLoad()) {
```
Comment (pieterlukasse) [inline src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java:154]: this if seems wrong as `importData` is always calling  `MySQLbulkLoader.flushAll();`

File: src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java [hunk 2]
```diff
@@ -135,8 +135,13 @@ public void run() {
...
 
-		    OptionSet options = ConsoleUtil.parseStandardDataAndMetaOptions(args, description, false);
-		    String dataFile = (String) options.valueOf("data");
+		    OptionSet options = ConsoleUtil.parseStandardDataAndMetaOptions(args, description, true);
+			if (options.has("loadMode") && !"bulkLoad".equals(options.valueOf("loadMode"))) {
+				throw new UnsupportedOperationException("This loader supports bulkLoad load mode only, but "
+						+ options.valueOf("loadMode")
+						+ " has been supplied.");
+			}
+			String dataFile = (String) options.valueOf("data");
```
Comment (pieterlukasse) [inline src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java:144]: was this message meant for src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java ?

File: src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java (hunk 1 above)
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/scripts/ImportCopyNumberSegmentData.java:154]: I tried to improve it here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/e3ec5d6064c9a7b3b0b19f19e28a674738ce4469

File: src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java (hunk 2 above)
Comment (forus) [reply src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java:144]: see here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/95c32f8eb3abb84c2d6d2deb9932dfb072c31786 and here https://github.com/cBioPortal/cbioportal-core/pull/47/commits/fc84a410c540be471258697a746b598d507246b8

Comment (pieterlukasse) [reply src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java:144]: I see, apparently https://github.com/cBioPortal/cbioportal-core/blob/main/src/main/java/org/mskcc/cbio/portal/scripts/ImportTimelineData.java is also always bulkload mode.

**Review by pieterlukasse — APPROVED** [review]

**Review by haynescd — APPROVED** [review]
LGTM

**Pull request merged**